src/drivers/redis.ts

```diff
@@ -75,6 +87,42 @@ export default defineDriver((opts: RedisOptions) => {
         await getRedisClient().set(p(key), value);
       }
     },
+    async getItemRaw(key) {
+      if (opts.saveRawAsBinary) {
```

Comment:
I think in next major we should unconditionally do this.  We can optionally add `getItemRaw` and `setItemRaw` keys in current version. 

(if it is too complex description, please simplify your changes to have `saveRawAsBinary` behavior without flag I can push other reactors to finalize)

Reply:
> We can optionally add getItemRaw and setItemRaw keys in current version. (as a result, driver doesn't need to directly do fallback base64 handling, core does)

This approach sounds much better to me, I'll try it out.

> I think in next major we should unconditionally do this. 

I would also favor this. My only concern is that it would require user action to make existing data accessible. But with clear enough migration notes it's probably ok. Perhaps a dev warning could also be added in a future minor release?

Reply:
Unstorage (core) can also be smart for raw operations returning `string` starting with a signature to reverse it so we can remain backward compatible.

BTW also consider `raw` methods are still marked as experimental (and that's partially why for nitro cache I didn't adopt raw to minimize affected users for such changes).

Reply:
I've updated the PR to only add `getItemRaw`/`setItemRaw` when the option is enabled and to add some tests for the behavior. 

I think we need to know upfront whether to expect binary or string data because that determines whether we call `ioredis.get` or `ioredis.getBuffer`. There may still be a way to catch a mismatch, but it's a good point that raw is still experimental, and there is a migration path.